designs/sam-debugging/local-sam-debugging.md

```diff
@@ -203,8 +203,8 @@ The required fields are: type, request, invokeTarget
                 "containerBuild": false,
                 "skipNewImageCheck": false,
                 "dockerNetwork": "aaaaa",
-                "buildArguments": "--foo",
-                "localArguments": "--foo",
+                "buildArguments": ["--foo"],
```

Comment:
The initial design was modeled after how this would be represented in a UI-rich environment like the JB Toolkit (a Text box). Interested in hearing if the motivation was idiomatic-driven, and more importantly, if it is any more or less user-friendly.

Reply:
args in a string lead to lots of platform-specific quoting issues. And users will think it's our bug.


If we start with string and get a  lot of user  reports, it's hard to go back and fix all of the user configs (we can't ever remove support for the single-string form). Whereas if  we start with an array of strings, and get  a lot of demand for the "sloppy" single-string form, we can add it later.


> how this would be represented in a UI-rich environment like the JB Toolkit (a Text box)

UI-rich approach hopefully offers more than just a textbox, since that is just the same thing as a string in a JSON  file.